cli/azd/pkg/auth/manager.go

```diff
@@ -0,0 +1,181 @@
+	if err != nil {
+		log.Println(err)
+		return nil, nil, nil, fmt.Errorf("fetching current user: %w", err)
+	}
+
+	currentUserHomeId, has := cfg.Get(cCurrentUserHomeIdKey)
+	if !has {
+		return nil, nil, nil, ErrNoCurrentUser
+	}
+
+	for _, account := range m.client.Accounts() {
+		if account.HomeAccountID == currentUserHomeId.(string) {
+			cred := m.newCredential(&account)
+			if tok, err := cred.GetToken(ctx, policy.TokenRequestOptions{Scopes: cLoginScopes}); err != nil {
+				return nil, nil, nil, fmt.Errorf("failed to get token: %v: %w", err, ErrNoCurrentUser)
+			} else {
+				return &account, m.newCredential(&account), &tok.ExpiresOn, nil
+			}
+		}
+
+		log.Printf("ignoring cached account with home id '%s', does not match '%s'",
+			account.HomeAccountID, currentUserHomeId.(string))
+	}
+
+	log.Println("got to end")
+	return nil, nil, nil, ErrNoCurrentUser
+}
+
+func (m *Manager) Login(
+	ctx context.Context,
+	useDeviceCode bool,
+) (*public.Account, azcore.TokenCredential, *time.Time, error) {
+
+	var authResult public.AuthResult
+
+	if useDeviceCode {
+		code, err := m.client.AcquireTokenByDeviceCode(ctx, cLoginScopes)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		fmt.Fprintln(m.out, code.Result.Message)
+
+		res, err := code.AuthenticationResult(ctx)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		authResult = res
+	} else {
+		res, err := m.client.AcquireTokenInteractive(ctx, cLoginScopes)
+		if err != nil {
+			return nil, nil, nil, err
+		}
+
+		authResult = res
+	}
+
+	cfg, err := config.GetUserConfig(m.configManager)
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("fetching current user: %w", err)
+	}
+
+	if err := cfg.Set(cCurrentUserHomeIdKey, authResult.Account.HomeAccountID); err != nil {
+		return nil, nil, nil, fmt.Errorf("setting account id in config: %w", err)
+	}
+
+	userConfigFilePath, err := config.GetUserConfigFilePath()
+	if err != nil {
+		return nil, nil, nil, fmt.Errorf("failed getting user config file path. %w", err)
+	}
+
+	if err := m.configManager.Save(cfg, userConfigFilePath); err != nil {
+		return nil, nil, nil, fmt.Errorf("failed saving configuration: %w", err)
+	}
+
+	log.Printf("logged in as %s (%s)", authResult.Account.PreferredUsername, authResult.Account.HomeAccountID)
+
+	return &authResult.Account, m.newCredential(&authResult.Account), &authResult.ExpiresOn, nil
+}
+
+func (m *Manager) Logout(ctx context.Context) error {
+	act, _, _, err := m.GetSignedInUser(ctx)
+	if errors.Is(err, ErrNoCurrentUser) {
+		// already signed out, nothing to do.
+		return nil
+	} else if err != nil {
+		return fmt.Errorf("fetching current user: %w", err)
+	}
+
+	// Unset the current user from config, but if we fail to do so, don't fail the overall operation
+	if cfg, err := config.GetUserConfig(m.configManager); err != nil {
+		log.Printf("error fetching config for current user during logout. ignoring: %v", err)
+	} else {
+		if err := cfg.Unset(cCurrentUserHomeIdKey); err != nil {
```

Comment:
You'll need to save the config with the `config.Manager` after any any `Set` or `Unset` operations to persist them back to the file system.

Reply:
Yup - thanks!